src/CMakeLists.txt

```diff
@@ -20,12 +25,18 @@ if (NOT _ONEDPL_PSTL_OFFLOAD_BINARY_PATH)
 
     add_library(pstloffload SHARED ${CMAKE_CURRENT_SOURCE_DIR}/pstl_offload.cpp)
 
+    if (CMAKE_HOST_WIN32)
+        set(EXTRA_COMPILE_FLAGS "/GS")
```

Comment:
In the docs for icx, it lists `/GS` the default option.  Have you found that this is not the case?

Reply:
I might be missing something. It seems /GS- is [default](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-1/gs-001.html).

Reply:
Interesting... I see it is the same in their 2024.0 guide as well.
However, from my windows environment:
`Intel(R) oneAPI DPC++/C++ Compiler 2024.0.0 (2024.0.0.20231017)`
excerpt from `icx /help`:
```
  /GS-                    Disable buffer security check
  /GS                     Enable buffer security check (default)
```

In any case, lets keep the flag there in this PR.  Thanks.